airbyte-integrations/connectors/source-harvest/unit_tests/integration/test_clients.py

```diff
@@ -0,0 +1,460 @@
+        Then: The connector should return zero records without errors
+        """
+        config = ConfigBuilder().with_account_id(_ACCOUNT_ID).with_api_token(_API_TOKEN).build()
+
+        # ARRANGE: Mock empty response
+        http_mocker.get(
+            HarvestRequestBuilder.clients_endpoint(_ACCOUNT_ID, _API_TOKEN)
+            .with_per_page(50)
+            .with_updated_since("2021-01-01T00:00:00Z")
+            .build(),
+            HarvestPaginatedResponseBuilder.empty_page("clients"),
+        )
+
+        # ACT: Run the connector
+        source = get_source(config=config)
+        catalog = CatalogBuilder().with_stream(_STREAM_NAME, SyncMode.full_refresh).build()
+        output = read(source, config=config, catalog=catalog)
+
+        # ASSERT: No records but no errors
+        assert len(output.records) == 0
+
+        # ASSERT: Should have log messages indicating successful sync completion
+        log_messages = [log.log.message for log in output.logs]
+        assert any("Finished syncing" in msg for msg in log_messages)
+
+    @HttpMocker()
+    def test_unauthorized_error_handling(self, http_mocker: HttpMocker):
+        """
+        Test that connector ignores 401 authentication errors and completes sync successfully.
+
+        The manifest configures 401 errors with action: IGNORE, which means the connector
+        silently ignores auth failures and continues the sync, marking it as successful
+        with 0 records rather than failing the sync.
+
+        Given: Invalid API credentials
+        When: Making an API request that returns 401
+        Then: The connector should ignore the error, return 0 records, and complete successfully
+        """
+        config = ConfigBuilder().with_account_id(_ACCOUNT_ID).with_api_token("invalid_token").build()
+
+        # ARRANGE: Mock 401 Unauthorized response
+        http_mocker.get(
+            HarvestRequestBuilder.clients_endpoint(_ACCOUNT_ID, "invalid_token")
+            .with_per_page(50)
+            .with_updated_since("2021-01-01T00:00:00Z")
+            .build(),
+            HttpResponse(body=json.dumps({"error": "invalid_token", "error_description": "The access token is invalid"}), status_code=401),
+        )
+
+        # ACT: Run the connector (401 errors are ignored, not raised)
+        source = get_source(config=config)
+        catalog = CatalogBuilder().with_stream(_STREAM_NAME, SyncMode.full_refresh).build()
+        output = read(source, config=config, catalog=catalog, expecting_exception=False)
+
+        # ASSERT: Sync completes successfully with 0 records (401 is ignored per manifest config)
+        assert len(output.records) == 0
+
+        # ASSERT: Should have the expected error message in logs
+        log_messages = [log.log.message for log in output.logs]
+        expected_msg = "Please ensure your credentials are valid."
+        assert any(expected_msg in msg for msg in log_messages), f"Expected auth error message in logs"
```

Comment:
Similar to the above, we should avoid performing any assertions on log messages. 

Instead of asserting on the presence of log messages, my recommendation is that we instead just assert that we did not receive any error messages in the output.

If the IGNORE behavior was not functioning properly, then we would end up with an explicit error message surfaced. So us checking that no errors are surfaced should be a sufficient assertion for this test